catalog/rocks/revparse.go

```diff
@@ -0,0 +1,76 @@
+package rocks
+
+import (
+	"regexp"
+	"strconv"
+)
+
+var (
+	hashRegexp      = regexp.MustCompile("^[a-fA-F0-9]{1,64}$")
+	modifiersRegexp = regexp.MustCompile("(^|[~^])[^^~]*")
```

Comment:
Not sure I understand.  I think you don't need capture here, so
```suggestion
	modifiersRegexp = regexp.MustCompile("(?:"^|[~^])[^^~]*")
```
would make it explicit?

Reply:
not sure what you meant to do here.. but I count 3 unescaped `"`'s, so I don't think it'll compile :)